www/docs/client/links/httpSubscriptionLink.md

```diff
@@ -178,8 +178,10 @@ If the client and server are not on the same domain, you can use `withCredential
 // [...]
 unstable_httpSubscriptionLink({
   url: 'https://example.com/api/trpc',
-  eventSourceOptions: {
-    withCredentials: true, // <---
+  eventSourceOptions() {
```

Comment:
I've removed the object variant of this because that way there is 0 nuance in when you can use it. Using a callback is a very small breaking change and gives tRPC more flexibility in calling it for newer values later

Reply:
I don't see the motivation for it - what's the problem with `resultOf(opts.eventSourceOptions)`? 👀

Reply:
We're still using resultOf internally

If the goal of shouldReinitialize is to get the latest copy of eventSourceOptions, then passing an object implicitly makes that functionality useless. Only allowing a callback means we don't need a load of conditional docs "use x if you're doing y, unless z then use f" becomes "use x if you want it, y if you need it"

Reply:
Alternative would be a warning that if shouldReinitializeOnError is set, tRPC moans that eventSourceOptions needs to be a callback?